docs/sources/reference/components/otelcol.exporter.awss3.md

```diff
@@ -0,0 +1,190 @@
+The following blocks are supported inside the definition of
+`otelcol.exporter.awss3`:
+
+Hierarchy              | Block                | Description                                                                          | Required
+-----------------------|----------------------|--------------------------------------------------------------------------------------|---------
+s3_uploader            | [s3_uploader][]      | Configures the AWS S3 bucket details to send telemetry data to.                      | yes
+marshaler              | [marshaler][]        | Marshaler used to produce output data.                                               | no
+debug_metrics          | [debug_metrics][]    | Configures the metrics that this component generates to monitor its state.           | no
+
+[s3_uploader]: #s3_uploader-block
+[marshaler]: #marshaler-block
+[debug_metrics]: #debug_metrics-block
+
+### s3_uploader block
+
+The `s3_uploader` block configures the AWS S3 bucket details used by the component.
+
+The following arguments are supported:
+
+Name                  | Type                       | Description                                                                      | Default      | Required
+----------------------|----------------------------|----------------------------------------------------------------------------------|--------------|---------
+`region`              | `string`                   | AWS region.                                                                      | `"us-east-1"`| no
+`s3_bucket`           | `string`                   | S3 bucket                                                                        |              | yes
+`s3_prefix`           | `string`                   | Prefix for the S3 key (root directory inside bucket).                            |              | yes
+`s3_partition`        | `string`                   | Time granularity of S3 key: hour or minute                                       | `"minute"`   | no
+`role_arn`            | `string`                   | The Role ARN to be assumed                                                       |              | no
+`file_prefix`         | `string`                   | File prefix defined by user                                                      |              | no
+`endpoint`            | `string`                   | Overrides the endpoint used by the exporter instead of constructing it from `region` and `s3_bucket` |      | no
+`s3_force_path_style` | `boolean`                  | [Set this to `true` to force the request to use path-style addressing](http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html) | `false`             | no
+`disable_ssl`         | `boolean`                  | Set this to `true` to disable SSL when sending requests           |              | `false`
+`compression`         | `string`                   | should the file be compressed                                                    | `none`      | no
```

Comment:
```suggestion
`region`              | `string`                   | The AWS region.                                                                      | `"us-east-1"`| no
`s3_bucket`           | `string`                   | The S3 bucket.                                                                        |              | yes
`s3_prefix`           | `string`                   | Prefix for the S3 key (root directory inside the bucket).                            |              | yes
`s3_partition`        | `string`                   | Time granularity of S3 key: hour or minute.                                       | `"minute"`   | no
`role_arn`            | `string`                   | The Role ARN to be assumed.                                                       |              | no
`file_prefix`         | `string`                   | The file prefix defined by the user.                                                      |              | no
`endpoint`            | `string`                   | Overrides the endpoint used by the exporter instead of constructing it from `region` and `s3_bucket`. |      | no
`s3_force_path_style` | `boolean`                  |  Set this to `true` to force the request to use [path-style requests](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#path-style-access) | `false`             | no
`disable_ssl`         | `boolean`                  |  Set this to `true` to disable SSL when sending requests           |              | `false`
`compression`         | `string`                   | should the file be compressed                                                    | `none`      | no

[path-style requests]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#path-style-access
```

Added a bunch of missing periods/full stops.

I've updated the external doc reference URL for path style addressing and the link text - is this still correct?

The description for `compression` doesn't look complete to me. It implies a boolean by the description, but it's a string. What do you put in the string? A compression algo? (zip, rar, tgz, etc?)

Reply:
I'll get the description cleaned up for `compression`. Just pulled from the OTEL docs. But that's a good call out.